compose/service.py

```diff
@@ -448,11 +479,26 @@ def recreate_container(self,
             else:
                 raise
 
+    def _recreate_rename_container(self, container):
         # Use a hopefully unique container name by prepending the short id
         self.client.rename(
             container.id,
-            '%s_%s' % (container.short_id, container.name))
+            '%s_%s' % (container.short_id, container.name)
+        )
```

Comment:
I think the `Container` class would be a better place for this new function. It only uses `self.client` and `container` (which are both available in `Container`).

Maybe something like `def rename_to_tmp_name(self):` ?